common/src/PCLPointCloud2.cpp

```diff
@@ -40,24 +40,25 @@
 
 #include <numeric>
 #include <vector>
+#include <cassert>
 
 #include <pcl/common/io.h>
 #include <pcl/pcl_macros.h>
 #include <pcl/exceptions.h>
 #include <pcl/PCLPointCloud2.h>
 
 bool
-pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPointCloud2 &cloud2)
+pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPointCloud2 &cloud2) noexcept
```

Comment:
This can't be `noexcept` since there are memory allocations being done inside the function

Reply:
Well, if  ``bad_alloc`` is thrown... there is very little we can do about it, [right](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-noexcept) ?

Reply:
Yes, but I don't think that makes the function noexcept, since it'll invoke `std::terminate` and remove the possibility of any RAII that the user might want to do (eg: lock files, kernel semaphores).

It's my personal opinion that we shouldn't assume our code is the only one using any resources. But OTOH I agree that `bad_alloc` is tough to handle. I'll refrain from making a decision on this hunk.

Reply:
Not sure what you meant here...
when ``std::terminate`` is invoked still destructors are [called](https://godbolt.org/z/egCBbx)
Did I answer your question?

Reply:
> when std::terminate is invoked still destructors are called

That's not standard. It might happen, it might not. It's implementation specific.